src/messaging/ExchangeContext.cpp

```diff
@@ -689,5 +698,21 @@ void ExchangeContext::ExchangeSessionHolder::GrabExpiredSession(const SessionHan
     GrabUnchecked(session);
 }
 
+bool ExchangeContext::IsPeerLikelyActiveHint()
+{
+    // If we are not the initiator, it means the peer sent the first message.
+    // This means our peer is active if we are sending a message to them.
+    if (!IsInitiator())
+        return true;
+
+    // If we create an Ephemeral Exchange, it was just to generate a StandaloneAck.
+    // Since we are sending an ack, we know the peer is active.
+    if (IsEphemeralExchange())
+        return true;
```

Comment:
By the way, I am pretty sure this code is unreached in this case (because acks are sent without reliability enabled for them), but it's probably OK to have this in here in case something else starts using this method.

Reply:
Also, IIUC, Ephemeral exchange is created to send only the ACk and then immediately closed. There is no other message that is meant to be sent on that exchange, correct?

Reply:
Correct.

Reply:
Correct, i added it in case it gets used for another purpose. I can remove it if you prefer.

Reply:
If this code is unreached for ephemeral exchanges right now, I think it would be better to `VerifyOrDie(!IsEphemeralExchange())` with a comment about how if we ever start using this for ephemeral exchanges we should at that point decide how those should behave.

Reply:
resolving comment since it is no longer applicable